src/chainstate/burn/db/sortdb.rs

```diff
@@ -2254,13 +2397,85 @@ impl <'a> SortitionHandleTx <'a> {
                     keys.push(db_keys::pox_last_anchor().to_string());
                     values.push("".to_string());
                 }
+                // if we've selected an anchor _and_ know of the anchor,
+                //  write the reward set information
+                if let Some(mut reward_set) = reward_info.known_selected_anchor_block_owned() {
+                    if reward_set.len() > 0 {
+                        // if we have a reward set, then we must also have produced a recipient
+                        //   info for this block
+                        let mut recipients_to_remove: Vec<_> = recipient_info.unwrap().recipients.iter().map(
+                            |(addr, ix)| (addr.clone(), *ix)).collect();
+                        recipients_to_remove.sort_unstable_by(|(_, a), (_, b)| b.cmp(a));
+                        // remove from the reward set any consumed addresses in this first reward block
+                        for (addr, ix) in recipients_to_remove.iter() {
+                            assert_eq!(&reward_set.remove(*ix as usize), addr,
+                                       "BUG: Attempted to remove used address from reward set, but failed to do so safely");
+                        }
+                    }
+
+                    keys.push(db_keys::pox_reward_set_size().to_string());
+                    values.push(db_keys::reward_set_size_to_string(reward_set.len()));
+                    for (ix, address) in reward_set.iter().enumerate() {
+                        keys.push(db_keys::pox_reward_set_entry(ix as u16));
+                        values.push(address.to_string());
+                    }
+                } else {
+                    keys.push(db_keys::pox_reward_set_size().to_string());
+                    values.push(db_keys::reward_set_size_to_string(0));
+                }
+
+                // in all cases, write the new PoX bit vector
                 keys.push(db_keys::pox_identifier().to_string());
                 values.push(pox_id.to_string());
+            } else {
+                // if this snapshot consumed some reward set entries AND
+                //  this isn't the start of a new reward cycle,
+                //   update the reward set
+                if let Some(reward_info) = recipient_info {
```

Comment:
What happens if `recipient_info` is `None` here?

Reply:
Any of:

* The current reward cycle had an anchor block, but it isn't known by this node.
* The current reward cycle did not have anchor block
* The current Stacking recipient set is empty (either because this reward cycle has already exhausted the set of addresses or because no one every Stacked).

Reply:
Added rustdocs in 45eb5cc06